### PR TITLE
dev-1600-create pytest indexd plugin

### PR DESCRIPTION
**Pull request by qqiao2024**
- [x] close socket after test for mock indexd server
- [x] add pytest-xdist in travis
- [x] rename indexd_test_utils2 to pytest_indexd. The utils is now a pytest plugin. We no longer need to import anything in conftest.  
https://jira.opensciencedatacloud.org/browse/DEV-1600

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -37,14 +30,30 @@ jobs:
...
+      before_script:
+        # Copy custom configs from the repo because PG-13 isn't set up to run like
+        # it normally does on Travis out of the box.
+        - sudo cp travis/postgresql.conf /etc/postgresql/13/main/postgresql.conf
+        - sudo cp travis/pg_hba.conf /etc/postgresql/13/main/pg_hba.conf
+        - sudo pg_ctlcluster 13 main restart
       script:
         - tox
-    - name: test utils2
+    - name: test utils2 noproc
```
Comment (stilesj-uchicago) [inline .travis.yml:41]: What is 'proc' versus 'noproc'?

File: indexd_test_utils2/__init__.py [hunk 2]
```diff
@@ -187,17 +171,18 @@ def indexd_server(pg_url: str) -> MockServer:
...
     host = os.getenv("INDEXD_HOST") or "localhost"
 
-    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
     debug = False
-
     port = random.randint(8000, 9000)
-    while sock.connect_ex((host, port)) == 0:
-        port = random.randint(8000, 9000)
+
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
```
Comment (stilesj-uchicago) [inline indexd_test_utils2/__init__.py]: You could probably extract this to a method or generator to yield a port.

File: indexd_test_utils2/__init__.py [hunk 3]
```diff
@@ -187,17 +171,18 @@ def indexd_server(pg_url: str) -> MockServer:
...
     """
     app = flask.Flask("indexd")
+    # the side effect of the following line creates the db and tables.
     settings = indexd_settings.get_settings(pg_url)
     app_init(app, settings)
 
     host = os.getenv("INDEXD_HOST") or "localhost"
 
-    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
     debug = False
```
Comment (stilesj-uchicago) [inline indexd_test_utils2/__init__.py]: Debug is always false. Maybe pull this out to an environment variable

File: tox.ini [hunk 4]
```diff
@@ -30,3 +32,19 @@ commands=
     env
     pip install . --force --upgrade
     python -m pytest {posargs: -lvvvs tests} --workers auto
+
+[testenv:xdist]
+setenv =
+    USE_TEST_UTILS2 = true
+;    mac
```
Comment (stilesj-uchicago) [inline tox.ini]: Would you please elaborate on this comment e.g., `MacOS requires the following environment variable`

**Review by stilesj-uchicago — APPROVED** [review]

File: .travis.yml (hunk 1 above)
Comment (qqiao2024) [reply .travis.yml:41]: Proc means pytest-postgresql will spin up postgres process. 
Noproc means pytest-postgresql will use existing postgres instance.

Comment (stilesj-uchicago) [reply .travis.yml:41]: Can we document that in the file? Or another place capturing similar concepts?

File: pytest_indexd/README.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
+# Pytest Indexd
```
Comment (mpsolano) [inline pytest_indexd/README.md:1]: Great README, nice work!

**Review by mpsolano — APPROVED** [review]
Looks good!

**Pull request merged**